### ISSUE TITLE
[BUG] SKR v1.3 + MKS MINI12864 + onboard SD

### ISSUE THREAD

**Issue by marvin-sinister** [MarlinFirmware/Marlin#15814]
### Description
Version: bugfix-2.0.x

If SD card is configured as ONBOARD for SKR v1.3, the MKS MINI1864 (and possibly others) doesn't work. I managed to get it to work by modifying the pins file for SKR as follows:
```
      #if ENABLED(FYSETC_MINI_12864)
      #define DOGLCD_CS    P1_21
      #define DOGLCD_A0    P1_22
      ...
      #elif ENABLED(FYSETC_MINI_12864_2_1)
        #define NEOPIXEL_PIN    P1_18
      #endif
```

And using:
`#define FYSETC_MINI_12864_2_1`
in `Configuration.h`.
And:
`#define SDCARD_CONNECTION ONBOARD`
In `Configuration_adv.h`.

I suspect that the core of the issue is that by setting `SDCARD_CONNECTION` to `ONBOARD` marlin uses wrong SPI for LCD.

### Steps to Reproduce

1. Connect MKS MINI12864 to SKR v1.3
2. set `SDCARD_CONNECTION` to `ONBOARD`
3. enable `MKS_MINI_12864`

**Expected behavior:** 
LCD working

**Actual behavior:**
LCD has only backlight

[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/3811584/Configuration.zip)

**Comment by Evg33** [comment]
MKS Mini not FYSETC_MINI_12864_2_1.
You need MKS_MINI_12864.
```
// MKS MINI12864 with graphic controller and SD support
// https://reprap.org/wiki/MKS_MINI_12864
//
#define MKS_MINI_12864

```

SKR v1.3 + FYSETC_MINI_12864_2_1 (Neopixel) + onboard SD working.
I do not have MKS_MINI_12864 for testing.
![image](https://user-images.githubusercontent.com/7160738/68257592-ddf7c480-0044-11ea-8b95-c601f6df8705.png)

**Comment by thisiskeithb** [comment]
I just tested an MKS Mini 12864 on an SKR 1.3 and can confirm that it works fine if you define `SDCARD_CONNECTION LCD`. If you set it to `SDCARD_CONNECTION ONBOARD`, nothing shows up on the LCD.

I'll have to confirm the MKS Mini 12864's pinout and then make sure LCD conditionals are setting the correct pins.

**Comment by boelle** [comment]
so there is no issue here? if so @marvin-sinister please hit close below

**Comment by thisiskeithb** [comment]
> so there is no issue here?

There is:

>  If you set it to SDCARD_CONNECTION ONBOARD, nothing shows up on the LCD.

**Comment by boelle** [comment]
yes but that is not a fault with marlin, or am i mistaken`?

**Comment by thisiskeithb** [comment]
> yes but that is not a fault with marlin, or am i mistaken`?

It's a bug in the SKR 1.3 pins file or possibly SPI issue since nothing shows up on the LCD If you enable `SDCARD_CONNECTION ONBOARD` And `MKS_MINI_12864` with an SKR 1.3.

**Comment by Evg33** [comment]
I use only SDCARD_CONNECTION ONBOARD.
Tested with next screens:
FYSETC_MINI_12864_1.2 / 2.0 / 2.1
REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER
ULTIPANEL
All ok.
I don't have MKS_MINI_12864 for testing.

**Comment by Evg33** [comment]
https://reprap.org/wiki/MKS_MINI_12864

**Comment by Evg33** [comment]
https://github.com/MarlinFirmware/Marlin/pull/13710

**Comment by marvin-sinister** [comment]
Okay, by playing around a bit i managed to get it working (and hopefuly didn't break anything else in process). The fix is as follows:
In `Marlin/src/pins/lpc1768/pins_BIGTREE_SKR_V1.3.h` we need to define `DOGLED` pins that are missing:
```
      #if ENABLED(MKS_MINI_12864)
        #define DOGLCD_CS  P1_21
        #define DOGLCD_A0  P1_22
        #define DOGLCD_SCK   P0_15
        #define DOGLCD_MOSI  P0_18
        #define FORCE_SOFT_SPI
      #endif
```
And in `Marlin/src/HAL/HAL_LPC1768/u8g/u8g_com_HAL_LPC1768_sw_spi.cpp` we need to use same code as `FYSETC_MINI_12864`, therefore we replace:
```
  #if ENABLED(FYSETC_MINI_12864)
```
With:
```
  #if EITHER(FYSETC_MINI_12864, MKS_MINI_12864)
```
In both places, so on line number 133 and 161.

After those changes MKS MINI12864 works and SD card can be set as `ONBOARD` or `LCD`. I don't have setup to test `CUSTOM_CABLE`.

**Comment by thisiskeithb** [comment]
@marvin-sinister Awesome! Wanna submit a PR?

**Comment by Evg33** [comment]
@thisiskeithb If you want to create PR, then remove double FORCE_SOFT_SPI from fysetc section, please.
https://github.com/MarlinFirmware/Marlin/blob/bugfix-2.0.x/Marlin/src/pins/lpc1768/pins_BIGTREE_SKR_V1.3.h#L229-L234

This is my mistake.

**Comment by thisiskeithb** [comment]
@Evg33: If @marvin-sinister doesn't create a PR and include, I can.

**Comment by boelle** [comment]
@marvin-sinister should your PR fix this?

**Comment by marvin-sinister** [comment]
Fixed by #15836
